### PR TITLE
Feature: add frontend services for backend models

### PR DESCRIPTION
**Pull request by someguy8**
Change parameters in service and model files to match those in api files. 

"Revert Revert Merge branch" was due to accidentally reverting a file merge within my local machine from upstream.

### CODE REVIEW

File: dream-big-api/app/api/category_api.rb [hunk 1]
```diff
@@ -11,12 +11,14 @@ class CategoryApi < Grape::API
...
   desc 'Allow creation of a category'
   params do
+    requires :id, type: Integer, desc: 'Category ID'
     requires :name, type: String, desc: 'Category name'
     requires :description, type: String, desc: 'The description of the category'
   end
   post '/category' do
     category_parameters = ActionController::Parameters.new(params)
       .permit(
+        :id,
```
Comment (lindenhutchinson) [inline dream-big-api/app/api/category_api.rb]: You don't need an ID to create a category, as the ID is created by the database when the category is created.

**Review by lindenhutchinson — CHANGES_REQUESTED** [review]

File: dream-big-api/app/api/category_api.rb (hunk 1 above)
Comment (someguy8) [reply dream-big-api/app/api/category_api.rb]: Thank you for the feedback. ID parameter has been removed from category_api.rb

File: dream-big-api/app/api/category_api.rb [hunk 2]
```diff
@@ -9,8 +9,8 @@ class CategoryApi < Grape::API
     present result, with: Entities::CategoryEntity
   end
 
-  desc 'Allow creation of a category'
```
Comment (Josh-Piper) [inline dream-big-api/app/api/category_api.rb]: Why did you remove the description of the API here? If it was by accident, can it be added back in.

Comment (someguy8) [reply dream-big-api/app/api/category_api.rb]: My apologies, that was an accident. I don't know why, but it seems I deleted the wrong line.
